phabricatoremails/render/template.py

```diff
@@ -0,0 +1,290 @@
+        the "templates" directory
+        """
+
+        return Template(
+            self._css_inline,
+            self.html_jinja_env.get_template(f"{template_name}.html.jinja2"),
+            self.text_jinja_env.get_template(f"{template_name}.text.jinja2"),
+        )
+
+
+def _jinja_html(loader):
+    jinja_env = jinja2.Environment(
+        loader=loader,
+        autoescape=True,
+        # If a template uses a variable that's not defined, throw an error
+        undefined=jinja2.StrictUndefined,
+        # the templates won't be changed at runtime, don't watch for changes
+        auto_reload=False,
+        # Cache all templates (rather than cycling the number of cached ones)
+        cache_size=-1,
+    )
+    jinja_env.tests["reply"] = _is_reply
+    jinja_env.tests["code"] = _is_code
+    jinja_env.tests["accepted"] = _is_reviewer_accepted
+    jinja_env.tests["blocking"] = _is_reviewer_blocking
+    jinja_env.tests["soft_needed"] = _is_reviewer_soft_needed
+    jinja_env.tests["needed"] = _is_reviewer_needed
+    jinja_env.tests["commented_on"] = _is_commented_on
+    jinja_env.tests["commented_on_secure"] = _is_commented_on_secure
+    jinja_env.filters["date"] = _date
+    jinja_env.filters["diff_class"] = _diff_class
+    jinja_env.filters["diff_symbol"] = _diff_symbol
+    jinja_env.filters["reviewer_actionable_class"] = _reviewer_actionable_class
+    jinja_env.filters["reviewer_status_icon"] = _reviewer_status_icon
+    jinja_env.filters["existence_change_class"] = _existence_change
+    jinja_env.filters["existence_change_label"] = _existence_change
+    jinja_env.filters["file_change"] = _file_change
+    jinja_env.filters["comment_summary"] = _comment_summary
+    jinja_env.filters["secure_comment_summary"] = _secure_comment_summary
+    jinja_env.globals["emoji"] = _emoji_html
+    return jinja_env
+
+
+def _jinja_text(loader):
+    jinja_env = jinja2.Environment(
+        loader=loader,
+        autoescape=False,  # These are text emails, we want raw characters
+        # If a template uses a variable that's not defined, throw an error
+        undefined=jinja2.StrictUndefined,
+        # the templates won't be changed at runtime, don't watch for changes
+        auto_reload=False,
+        # Cache all templates (rather than cycling the number of cached ones)
+        cache_size=-1,
+    )
+    jinja_env.tests["accepted"] = _is_reviewer_accepted
+    jinja_env.tests["blocking"] = _is_reviewer_blocking
+    jinja_env.tests["soft_needed"] = _is_reviewer_soft_needed
+    jinja_env.tests["needed"] = _is_reviewer_needed
+    jinja_env.filters["comment"] = _text_comment
+    jinja_env.filters["reviewer_status"] = _text_reviewer_status
+    jinja_env.filters["existence_change"] = _text_existence_change
+    jinja_env.filters["file_change"] = _file_change
+    jinja_env.filters["comment_summary"] = _comment_summary
+    jinja_env.filters["secure_comment_summary"] = _secure_comment_summary
```

Comment:
Names here can probably be a little more consistent, similar to what you have above.

Reply:
I'm not sure what you're referring to: if you mean that some of them have the `_text` prefix, that's because the behaviour of the filter was different between the text and html template.

For the rest of them, since they're shared functions, they shouldn't have the prefix.

If that's not what you're referring to, could you elaborate?